tests/test_misc.py

```diff
@@ -9,6 +10,16 @@ def test_cache_fn_url(self):
         url = "http://repo.continuum.io/pkgs/pro/osx-64/"
         self.assertEqual(cache_fn_url(url), '7618c8b6.json')
 
+    def test_ascii2sig(self):
+        self.assertEqual(sig2ascii(1234), 'BNI=')
+
+    def test_sig2ascii(self):
+        self.assertEqual(ascii2sig('BNI='), 1234)
+
+    def test_sig2ascii2sig(self):
+        for i in range(10000):
+            self.assertEqual(ascii2sig(sig2ascii(i)), i)
+
```

Comment:
Please also add corner case/out of bounds tests such as:
* negative values
* nulls/NaNs/zero
* numerical but invalid types (floats, decimal, datetime)

The code should handle these gracefully by raising an appropriate exception.  At least, the tests would verify that this code does not silently propagate invalid parameters and behavior under these circumstance.

(The reason I ask for these is because the concept of a "signature" has varied form and not everyone necessarily knows what to expect as the proper type of value (or range of values) for a signature.)